runtime/interpreter/value.go

```diff
@@ -1342,6 +1342,25 @@ func (v *StringValue) GetMember(interpreter *Interpreter, locationRange Location
 				return v.ToLower(invocation.Interpreter)
 			},
 		)
+
+	case sema.StringTypeReplaceAllFunctionName:
+		return NewHostFunctionValue(
+			interpreter,
+			sema.StringTypeReplaceAllFunctionType,
+			func(invocation Invocation) Value {
+				of, ok := invocation.Arguments[0].(*StringValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
+
+				with, ok := invocation.Arguments[1].(*StringValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
```

Comment:
Can these two parameters be of type `CharacterValue` (other than `StringValue`)?
Maybe also add a test(s) for passing `Character` typed values for `of` and `with` parameters?
```cadence
let a: Character = "x"
let b: Character = "y"

"something".replaceAll(of: a, with: b)
```

Reply:
ok, I can see that Kotlin `replace` allows either Char or String. May be we can allow that too.

Reply:
Actually, what I meant / my question was, whether we can get a `Character` value at runtime somehow.
If not that's totally fine. Just having a test-case to verify would be good.

With respect to: whether we need to support `Character`s in `replaceAll`, I think that could be a separate discussion/feature. And might be good it add it (if we want to add) in a separate PR. The smaller the PR is the better :)

Reply:
Ah I see. No it is disallowed. Added a checker test case in 34067af33.

Reply:
nice!